packages/web-components/src/components/ic-data-row/ic-data-row.tsx

```diff
@@ -75,82 +71,75 @@ export class DataRow {
-   * @param label string - label of value
-   * @returns HTMLDivElement - returns label as slot or ic-typography with label as textContent
-   */
-  private renderLabel = (label: string) => {
-    if (isSlotUsed(this.el, "label")) {
-      return (
-        <div class="label">
-          <slot name="label"></slot>
-        </div>
-      );
-    } else if (label) {
-      return (
-        <div class="label">
+  private renderCellContent = (cell: "label" | "value") => {
+    const isValue = cell === "value";
+    const Component = isValue ? "dd" : "dt";
+    return (
+      <Component class={cell}>
+        {isSlotUsed(this.el, cell) ? (
+          <slot name={cell}></slot>
+        ) : (
           <ic-typography
-            variant={this.entitySize === "xs" ? "label" : "subtitle-large"}
+            variant={
+              isValue
+                ? "body"
+                : this.entitySize === "xs"
+                ? "label"
+                : "subtitle-large"
+            }
           >
-            {label}
+            {isValue ? this.value : this.label}
           </ic-typography>
-        </div>
-      );
-    }
-
-    return null;
+        )}
+      </Component>
+    );
   };
 
   private labelEndComponent(): void {
-    const component = this.el.shadowRoot.querySelectorAll(
-      "slot[name=end-component]"
-    );
-
-    component.forEach((child: HTMLElement) =>
-      child.setAttribute("aria-label", "for " + this.label + " row")
-    );
+    this.el.shadowRoot
+      .querySelectorAll("slot[name=end-component]")
+      .forEach((child) =>
+        child.setAttribute("aria-label", "for " + this.label + " row")
+      );
   }
 
   render() {
-    const { label, value, small, size } = this;
+    const {
+      el,
+      entitySize,
+      hasEndComponent,
+      label,
+      renderCellContent,
+      size,
+      small,
+      value,
+    } = this;
 
     return (
       <Host
         class={{
           ["small"]: small || size === "small",
-          ["breakpoint-medium"]: this.entitySize === "m",
-          ["breakpoint-xs"]: this.entitySize === "xs",
+          ["breakpoint-medium"]: entitySize === "m",
+          ["breakpoint-xs"]: entitySize === "xs",
         }}
         role="listitem"
       >
         <div class="data">
           <div class="text-cells">
-            {this.renderLabel(label)}
-            <div class="value">
-              <slot name="value">
-                <ic-typography variant="body">{value}</ic-typography>
-              </slot>
-            </div>
+            {(isSlotUsed(el, "label") || label) && renderCellContent("label")}
+            {(isSlotUsed(el, "value") || value) && renderCellContent("value")}
           </div>
-          {this.hasEndComponent && (
-            <div class="end-component">
-              <div role="cell">
-                <slot name="end-component"></slot>
-              </div>
-            </div>
+          {hasEndComponent && (
+            <dd class="end-component">
```

Comment:
Didn't realise the role was only on the last 'cell'. If the role or div was deleted, what affect would that make? I might be missing something. Just trying to understand if `role=cell` was not set on the other label and value divs, is it required for the end cell?

Reply:
I don't believe it was required for any of the cells, which is why I removed it. role="cell" is used in table setups, rather than lists of data like this, so it was being used incorrectly